### PR TITLE
Add map layers legends

### PR DESCRIPTION
**Pull request by barbara-chaves**
This PR adds the map layers legends

## Acceptance criteria

The designs of the legend can be found [here](https://www.figma.com/file/3epSYh4KQZBCyh4Y5OzmHw/HeCo---Design-System?node-id=3656%3A12756), there is an example of style per type of data.

### HeCo priority landscapes
Title only with small square in #3d6c50

### Sub-basin boundaries
Title only with small square in ​#2485bc

### Protected areas
Title only with small square in ​#bafdd1 (70% opacity)

### Biodiversity intactness
Gradient: #f8ebff (label "Low"), #eccafc, #dfa4ff, #c26dfe, #9d36f7, #6d00e1, #3c00ab (label "High")[RW data object](https://api.resourcewatch.org/v1/dataset/1ddacdc5-b05e-4ba0-91e6-9b99c66f70b2/layer)

### Tree biomass density
Gradient: #895122 (label “0”), #957f4f, #9db38a, #39a401 (label "480 t Ha⁻¹")[RW data object](https://api.resourcewatch.org/v1/dataset/587b9951-54aa-4cd5-8929-8aacd5743e15/layer)

### Wetlands
Title only with small square in ​#0d9dc9 (60% opacity)

### Population density
Choropleth: #32095d (label "≤25"), #781c6d (label "≤100"), #ba3655 (label "≤1k"), #ed6825 (label "≤5k"), #fbb318 (label "≤10k"), #fcfea4 (label ">10k")[RW data object](https://api.resourcewatch.org/v1/dataset/f6b81da7-a846-45c1-88b2-dad95a26e523/layer)

### Endangered species and critical habitats
Title only with small square in #b2d26e[RW data object](https://api.resourcewatch.org/v1/dataset/5b5a21ac-0835-43fb-86b9-64b93d472e10/layer/4b5140a2-28ea-4707-8998-a9d1c32dbf53)

### Tree cover loss
Title only with small square in #dc6c9a[RW data object](https://api.resourcewatch.org/v1/dataset/b584954c-0d8d-40c6-859c-f3fdf3c2c5df/layer)

### Riverine flood risk
Choropleth: #990000 (label "Extremely high"), #ff1900 (label "High"), #ff9900 (label "Medium to high"), #ffe600 (label "Low to medium"), #ffff99 (label "Low"), #808080 (label "Arid & low water use"), #4e4e4e (label "No data")[RW data object](https://api.resourcewatch.org/v1/dataset/5152c286-53c1-4583-9519-816a6e41889d/layer)

### Indigenous Reserves
Title only with small square in ​#ffb780 (70% opacity)

## Testing instructions

- In discover/project page, enable and disable the map layers
- The legends should be displayed at the bottom right of the map
- Be sure that the legends can be enabled and disabled properly

## Tracking

[LET-1054](https://vizzuality.atlassian.net/browse/LET-1054)

### CODE REVIEW

File: frontend/components/map/legend/item/component.tsx [hunk 1]
```diff
@@ -1,33 +1,40 @@
...
+            {icon}
+            <span className="text-xs leading-[14px]">{name}</span>
+          </div>
+        </div>
+        <Button
+          type="button"
+          className="absolute z-50 flex items-center w-3 h-3 right-2"
+          onClick={handleCloseLegend}
+          theme="naked"
+          icon={() => <X className="text-gray-800 transition-transform hover:rotate-180" />}
```
Comment (clementprdhomme) [inline frontend/components/map/legend/item/component.tsx:31]: The icon is barely visible. I would suggest you move the icon inside the button like this:
```jsx
<Icon
  icon={X}
  className="w-4 h-4 stroke-[3px] text-gray-800 transition-transform hover:rotate-180"
/>
```

The size should be `w-4 h-4` and you can increase the stroke-width using `stroke-[3px]`.

**Review by clementprdhomme — COMMENTED** [review]
Good job! It must have taken you some time to figure out how to implement the legend!

I've added some comments. Let me know if you need some help.

File: frontend/containers/discover-map/layer-legend/component.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    return itms;
+  }, [layersLegends, sortArray]);
+
+  const id = useId();
+
+  // Callbacks
+  const onChangeOrder = useCallback((ids) => {
+    setSortArray(ids);
+  }, []);
+  const [active, setActive] = useState(true);
```
[the remark this answers is not in the document]
Comment (barbara-chaves) [reply frontend/containers/discover-map/layer-legend/component.tsx:40]: Do we have a way to know the order of the layer on the map?

[the remark this answers is not in the document]
Comment (clementprdhomme) [reply frontend/containers/discover-map/layer-legend/component.tsx]: In [LET-1053](https://vizzuality.atlassian.net/browse/LET-1053), we'll only allow the user to display one layer per group at a time. I think we could have this order:
- Priority Layers (at the very top)
- Contextual Layers (in the middle)
- Base Layers (at the very bottom)

Then inside each group, we could follow the same order: if the layer is at the beginning of the list, it's at the bottom, and if it's at the end, it's at the top.

What do you think?

**Review by clementprdhomme — APPROVED** [review]
I've made tiny changes to your code. I think it's ready to be merged.

**Pull request merged**